composer.json

```diff
@@ -11,12 +11,12 @@
     ],
     "require": {
         "php":                 "^7.4.0",
-        "composer-plugin-api": "^1.1.0"
+        "composer-plugin-api": "^1.1.0 || ^2.0"
     },
     "require-dev": {
         "phpunit/phpunit":          "^9.0.1",
         "infection/infection":      "^0.15.3",
-        "composer/composer":        "^1.9.3",
+        "composer/composer":        "^1.9.3 || ^2.0@dev",
```

Comment:
Do you plan on maintaining 1.x and 2.x in parallel? Otherwise I would bump the dependency in general, and keep older versions compatible with 1.x instead

Reply:
Yeah, looking further at this, we should most likely support only one format at a time (with different versions).

Also important: `installed.json` should already be up-to-date (assuming `composer install`: I can still see `composer dump-autoload` failing if `vendor` is under VCS)

Reply:
I think it's probably best to keep both here until we hit 2.0 stable at least, but I don't mind too much what you set in require-dev. For composer-plugin-api though I would absolutely keep requiring both versions until end of the year at least, as I don't expect we'll get everyone to upgrade to composer 2 within a week :(

Reply:
That said we do support php5.3+ in v2 for the exact purpose of getting everyone migrated without excuses, so that we can drop v1 support ASAP, not sure when that will be though.

Reply:
> as I don't expect we'll get everyone to upgrade to composer 2 within a week :(

Indeed, but since there will be a new minor, SAT shouldn't have an issue here, right?

Reply:
Given the code "overhead" is one extra if/line I'd still support both composer versions/formats in one version here, it's much simpler and it will help keep things sane for teams with people who did not all migrate to composer v2 yet.

Reply:
I think I'll schedule removal of `1.x` on `1.9.0` in a separate patch then :+1:

Reply:
CI does indeed fail on uncovered mutations: https://travis-ci.org/github/Ocramius/PackageVersions/jobs/670142351

Will inspect on weekend, putting it on calendar.

Reply:
Moved to #135